diff --git a/dist/main/GoTrueClient.js b/dist/main/GoTrueClient.js
index 11125ff7186f75970090a44b9c8789cc61d8d95d..fc20b4e870dd5eebd0613431a5dfc5f5d0043334 100644
--- a/dist/main/GoTrueClient.js
+++ b/dist/main/GoTrueClient.js
@@ -52,7 +52,7 @@ class GoTrueClient {
         this.initializePromise = null;
         this.detectSessionInUrl = true;
         this.hasCustomAuthorizationHeader = false;
-        this.suppressGetSessionWarning = false;
+        this.suppressGetSessionWarning = true // PATCH: suppress warning for `getSession` by default
         this.lockAcquired = false;
         this.pendingInLock = [];
         /**
@@ -892,6 +892,10 @@ class GoTrueClient {
         }
         catch (error) {
             if ((0, errors_1.isAuthError)(error)) {
+                // PATCH: If the token holds an invalid session, remove it from storage
+                if (error.code === 'session_not_found') {
+                    await this._removeSession()
+                }
                 return { data: { user: null }, error };
             }
             throw error;
diff --git a/dist/module/GoTrueClient.js b/dist/module/GoTrueClient.js
index 353bee130187b1cc3d03f8be0c4bd5c296c5fc0b..1dfce305895571a08db9ffbeea07e64fa1806bc9 100644
--- a/dist/module/GoTrueClient.js
+++ b/dist/module/GoTrueClient.js
@@ -47,7 +47,7 @@ export default class GoTrueClient {
         this.initializePromise = null;
         this.detectSessionInUrl = true;
         this.hasCustomAuthorizationHeader = false;
-        this.suppressGetSessionWarning = false;
+        this.suppressGetSessionWarning = true // PATCH: suppress warning for `getSession` by default
         this.lockAcquired = false;
         this.pendingInLock = [];
         /**
@@ -887,6 +887,10 @@ export default class GoTrueClient {
         }
         catch (error) {
             if (isAuthError(error)) {
+                // PATCH: If the token holds an invalid session, remove it from storage
+                if (error.code === 'session_not_found') {
+                    await this._removeSession()
+                }
                 return { data: { user: null }, error };
             }
             throw error;
diff --git a/src/GoTrueClient.ts b/src/GoTrueClient.ts
index 88bedadcc615bea08e673ae5e79ad848578146ff..dcf71042adbd5898dceedd3618a1fbf2e546818d 100644
--- a/src/GoTrueClient.ts
+++ b/src/GoTrueClient.ts
@@ -156,7 +156,7 @@ export default class GoTrueClient {
     [key: string]: string
   }
   protected hasCustomAuthorizationHeader = false
-  protected suppressGetSessionWarning = false
+  protected suppressGetSessionWarning = true // PATCH: suppress warning for `getSession` by default
   protected fetch: Fetch
   protected lock: LockFunc
   protected lockAcquired = false
@@ -1163,6 +1163,8 @@ export default class GoTrueClient {
     return result
   }
 
+
+
   private async _getUser(jwt?: string): Promise<UserResponse> {
     try {
       if (jwt) {
@@ -1192,6 +1194,10 @@ export default class GoTrueClient {
       })
     } catch (error) {
       if (isAuthError(error)) {
+        // PATCH: If the token holds an invalid session, remove it from storage
+        if (error.code === 'session_not_found') {
+          await this._removeSession()
+        }
         return { data: { user: null }, error }
       }
 
